dali/pipeline/operator/operator.cc

```diff
@@ -23,7 +23,7 @@ void OperatorBase::EnforceUniformInputBatchSize(const workspace_t<Backend> &ws)
     DALI_ENFORCE(curr_batch_size == ws.GetInputBatchSize(i),
                  "Batch size has to be uniform across one iteration.");
   }
-  auto argument_ws = dynamic_cast<const ArgumentWorkspace &>(ws);
+  const auto &argument_ws = dynamic_cast<const ArgumentWorkspace &>(ws);
```

Comment:
Nice catch. Actually, this is a superclass, so this should work, too (and be faster):
```suggestion
  const ArgumentWorkspace &argument_ws = ws;
```